### PR TITLE
Replace colored printlns with tracing

### PR DESCRIPTION
**Pull request by sdankel**
Closes https://github.com/FuelLabs/sway/issues/1661

Replaces usages of `println` with coloring with calls to `tracing`. 

- The tracing subscriber is configured to choose stdout when the log level is ERROR or WARN
- I tested locally that `println_red_err` still logs to stderr and `println_red` still logs to stdout
- I removed some functions that were not used in any fuel repos ("print_*_err")
- Existing usages of `tracing:error` and `tracing:warn` now write to stderr; this caused some e2e tests to fail. Fixed this by making the vm e2e test harness look for expected output in both stderr and stdout.

### CODE REVIEW

File: forc-util/src/lib.rs [hunk 1]
```diff
@@ -201,65 +201,31 @@ pub fn print_on_failure(silent_mode: bool, warnings: &[CompileWarning], errors:
...
 pub fn println_green_err(txt: &str) {
-    println_std_err(txt, TermColor::Green);
+    println_std_err(txt, Colour::Green);
 }
 
-pub fn print_std_out(txt: &str, color: TermColor) {
-    let stdout = StandardStream::stdout(ColorChoice::Always);
-    print_with_color(txt, color, stdout);
+fn println_std_out(txt: &str, color: Colour) {
+    tracing::info!("{}", color.paint(txt));
```
Comment (mitchmindtree) [inline forc-util/src/lib.rs:224]: Omg, it was as easy as that all along :heart_eyes:

File: forc-util/src/lib.rs [hunk 2]
```diff
@@ -439,6 +431,7 @@ pub fn init_tracing_subscriber() {
         .with_line_number(false)
         .without_time()
         .with_target(false)
+        .with_writer(StdioTracingWriter)
```
Comment (mitchmindtree) [inline forc-util/src/lib.rs:434]: Nicccee

**Review by mitchmindtree — APPROVED** [review]
Awesome stuff @sdankel, especially for your first Rust PR! Hopefully the first of many :pray: 

:shipit: 

Also cc @otrho, just a heads up that we're now capturing both stdout and stderr in E2E test output - should be fine right?

**Review by JoshuaBatty — APPROVED** [review]
Great work Sophie!

**Pull request merged**